draft-jennings-moq-secure-objects.md

```diff
@@ -201,326 +190,248 @@ name but different data.
 One of the ways system keep the object names unique is by using a fully
 qualified domain names or UUIDs as part of the TrackNamespace.
 
+# Secure Objects
+
+A Secure MOQT Object is MOQT OBJECT_STREAM or OBJECT_DATAGRAM messages that has
+been protected with the scheme defined in this draft, so that its payload is
+encrypted and the whole message is authenticated by an authentication tag in the
+payload:
+
+~~~ aasvg
+   OBJECT_STREAM Message {
+ +-----------------------------------------+
+ |   Subscribe ID (i)        authenticated |
+ |   Track Alias (i)                       |
+ |   Group ID (i)                          |
+ |   Object ID (i)                         |
+ |   Object Send Order (i)                 |
+ |   Object Status (i)                     |
+ | +-------------------------------------+ |
+ | | Object Payload (..)       encrypted | |
+ | +-------------------------------------+ |
+ +-----------------------------------------+
+   }
+~~~
+{: #fig-encrypted-object title="Security properties of a secure OBJECT_STREAM
+object.  Protection of an OBJECT_DATAGRAM message is analogous." }
+
+Specifically, the payload of a secure object comprises the "encrypted data" and
+"authentication tag" portions of an SFrame ciphertext object, as described in
+{{Section 4.2 of !I-D.ietf-sframe-enc}}.  That is, the secure object payload is
+an SFrame ciphertext without the SFrame Header.  Rather than transmitting the
+SFrame Header, MOQT producers and consumers synthesize it from the information
+in the object being protected, as described in {{mapping-between-sframe-and-moqt}}.
+
+## Setup Assumptions
+
+This scheme applies SFrame on a per-track basis.  We assume that the application
+assigns each track a unique secret value that is used as the `base_key` for
+SFrame, where this value is known only to authorized producer and consumers for
+a given track.  How these per-track secrets are established is outside the scope
+of this specification.
+
+It is also up to the application to specify the ciphersuite to be used for each
+track's SFrame context.  Any SFrame ciphersuite can be used. 
+
```

Comment:
We need to use KID for both these rekey use case, the multiple sender use case, and probably others.  I think this needs to be changed back to include the KID in what is sent over the wire. Probably with a var int and convention that if it is zero, then use same KID from prior object with a KID.